### ISSUE TITLE
Internal server error (for small, negative longitudes?)

### ISSUE THREAD

**Issue by drmrbrewer** [alexander0042/pirate-weather-ha#9]
Firstly, I love this project... kudos both on the idea and the implementation!  Will be fascinating to see how it develops!

An issue I'm having at present is that the API chokes on the following call:

https://api.pirateweather.net/forecast/[api_key]/51.507,-0.128

I'm not 100% sure, but I *think* that this only happens for small, negative longitude values (below 0.8?) as per the following example calls:

https://api.pirateweather.net/forecast/[api_key]/51.5,-0.1 : internal server error
https://api.pirateweather.net/forecast/[api_key]/51.5,-0.2 : internal server error
https://api.pirateweather.net/forecast/[api_key]/51.5,-0.3 : internal server error
https://api.pirateweather.net/forecast/[api_key]/51.5,-0.7 : internal server error
https://api.pirateweather.net/forecast/[api_key]/51.5,-0.8 : OK!
https://api.pirateweather.net/forecast/[api_key]/0,-0.7 : internal server error
https://api.pirateweather.net/forecast/[api_key]/0,-0.8 : OK!

Hope you can shed some light, and thanks again for your work!

**Comment by alexander0042** [comment]
Thanks for the support! It's really great to see people using this, and I really appreciate the feedback. 

This seems to be a true "edge case", in every sense of the word. It's almost certainly an issue with how the code is picking the nearest grid point to a value, but let me dig into this and post back.

**Comment by alexander0042** [comment]
It was indeed- I thought I'd fixed all my "edge of grid" issues when I corrected for the lack of a sunset at the North Pole this time of year, but apparently not!

I'm using the 0.5 degree GFS model, so there's 721 longitudinal cells (360*2+1). The algorithm does an interpolation between the 9 closest cells to avoid sudden jumps in the forecast when moving between grids points; however, this doesn't work at the edges of the grid, where I'm just using the specific cell. The issue was that it was using 721 as the "edge of grid" index, instead of 720, so trying to retrieve the cells 720-722 (721-1, 720, 721+1). Since that doesn't exist, everything then broke.

I've pushed a new version of the Lambda function that corrects this and made a new test case with the location 89.99,-0.1001 to look for these issues in the future. 

Thanks again for pointing this out. I come from a MATLAB background (I originally started this project as a way to learn Python), and I always forget that Python arrays start at zero, so appreciate this catch! As a thank you, shoot me a e-mail to api@alexanderrey.ca if you want me to raise your api limit 🥇 

Thanks,
Alexander
